middlewares/prometheus_test.go

```diff
@@ -44,4 +46,24 @@ func TestPrometheus(t *testing.T) {
 	if !strings.Contains(body, latencyName) {
 		t.Errorf("body does not contain request duration entry '%s'", reqsName)
 	}
+
+	// Register the same metrics again
+	metricsMiddlewareBackend = NewMetricsWrapper(NewPrometheus("test", &types.Prometheus{}))
+	n = negroni.New()
+	n.Use(metricsMiddlewareBackend)
+	n.UseHandler(r)
+
+	recorder = httptest.NewRecorder()
+	n.ServeHTTP(recorder, req2)
+
+	body = recorder.Body.String()
+	re := regexp.MustCompile(`(?m)traefik_requests_total.*?(\d)$`)
```

Comment:
This test seems to be testing the implementation of the metrics handler. While I'm all in favor of doing that, I believe it should go into (the yet missing) `metrics_test.go`.

More importantly though, I'd probably test it differently by means of comparing the counter label name and value of the underlying `LabelPair`. Doing a regular expression to extract the number of total requests is somewhat brittle since we rely on an implementation detail of Prometheus that could change in a future version.

My proposal would be to remove this part of the test and leave it for another PR (which I'd be happy to do as well). I think the next test paragraph covers what we want to achieve with the PR at hand.